src/ngraph/op/util/rnn_cell_base.hpp

```diff
@@ -101,16 +99,15 @@ namespace ngraph
                 ///
                 /// \return     Node with element-wise multiply operation.
                 ///
-                static std::shared_ptr<Node> mul(const std::shared_ptr<Node>& lhs,
-                                                 const std::shared_ptr<Node>& rhs);
+                static std::shared_ptr<Node> mul(const Output<Node>& lhs, const Output<Node>& rhs);
                 ///
                 /// \brief      Creates node with element-wise clip operation with numpy broadcasting.
                 ///
                 /// \param[in]  data   The input tensor for clipping.
                 ///
                 /// \return     Node with element-wise clip operation.
                 ///
-                std::shared_ptr<Node> clip(const std::shared_ptr<Node>& data) const;
+                std::shared_ptr<Node> clip(const Output<Node>& data) const;
```

Comment:
`Output` type for input data is little confusing, isn't it?
Maybe it's worth adding an some kind of alias to it?

Reply:
An `Output` of an op provides a value for an `Input` of another op.

Reply:
@tsocha We've been chatting about this a little bit. I think the "Input/Output" nomenclature feels right when you are talking about the graph as a whole or doing rewrites post-construction, but I agree that when you are building the graph in an "applicative" style it sometimes reads a bit strangely. It might be a matter of getting people used to the new style, or maybe an alias is in order as you suggest.

Any thoughts on what a good alias/different name might be?

Reply:
It feels like this represents an edge of the graph, or a connection between 2 nodes. 
Perhaps a name like `Connection` or `Link` could work?

Reply:
I don't think `Connection` or `Link` (or, say, `Edge`) is quite right, because the object in question really only corresponds to an output, and has no affinity with any particular target input. It's used in other contexts to say things like `n->output(i).get_partial_shape()` (where `n->output(i)` is an `Output<Node>`).

I wonder if `Value` as an alias for `Output<Node>` might make sense? Maybe something like `InputSource`?

Reply:
The easiest way is to use `Input` as an alias but is this name already used?
`InputSource` is seems good as well.

Reply:
`Input` is already taken, yeah. (It refers to an "input port" of a node, and the distinction is important at later phases when we are rewriting graphs.)